src/controllers/portfolio/portfolio.ts

```diff
@@ -664,6 +726,91 @@ export class PortfolioController extends EventEmitter {
     this.emitUpdate()
   }
 
+  /**
+   * Learn new tokens from humanizer and debug_traceCall
+   * @param tokens: []
+   * @param storagePreviousHints: {}
+   * @param networkId: NetworkId
+   */
+  async learnTokens(tokens: string[], networkId: NetworkId) {
+    const storagePreviousHints = await this.#storage.get('previousHints', {})
+
+    const learnedTokens = storagePreviousHints.learnedTokens || {}
+    const networkLearnedTokens = learnedTokens[networkId] || {}
+
+    // Get the limit for the network, refactor this maybe, this seems like a hack
+    const keyWithNetwork = [...this.#portfolioLibs.keys()].find((k) => k.includes(networkId))
+
+    const limit = this.#portfolioLibs[keyWithNetwork]?.value.deploylessTokens.isLimitedAt24kbData
```

Comment:
This part looks weird.

Reply:
Yes, this part needs to be rethinked how to be changed